src/codegen/detail/code_gen_c.h

```diff
@@ -236,6 +245,31 @@ template <class Stream> void CodeGenC<Stream>::visit(const Store &op) {
     this->os() << ";" << std::endl;
 }
 
+template <class Stream> void CodeGenC<Stream>::visit(const Alloc &op) {
+    this->markUseBuffer(op->var_);
+    this->makeIndent();
+
+    // e.g. float (*x)[8][11] = new float[5][8][11];
+    auto &&tensor = BaseClass::buffer(op->var_)->tensor();
+    auto &&shape = tensor->shape();
+    auto &&dtype = tensor->dtype();
+
+    this->os() << mangle(op->var_);
+
+    this->os() << " = new " << gen(dtype);
+    for (auto &&dim : shape) {
+        this->os() << "[";
+        (*this)(dim);
+        this->os() << "]";
+    }
+    this->os() << ";" << std::endl;
+}
+
+template <class Stream> void CodeGenC<Stream>::visit(const Free &op) {
+    this->makeIndent();
+    this->os() << "delete " << mangle(op->var_) << ";" << std::endl;
```

Comment:
Do we need to distinguish `delete` for scalars and `delete[]` for non-scalars?

Reply:
I see `Alloc` and `Free` won't be inserted for scalars, please document it somewhere, perhaps in a doc-string for `makeHeapAlloc`. Then, do we need to change `delete` to `delete[]`?